api/v3/Activity.php

```diff
@@ -471,7 +471,7 @@ function _civicrm_api3_activity_get_formatResult($params, $activities, $options)
         $dao = CRM_Core_DAO::executeQuery("SELECT activity_id, case_id FROM civicrm_case_activity WHERE activity_id IN (%1)",
           array(1 => array(implode(',', array_keys($activities)), 'String', CRM_Core_DAO::QUERY_FORMAT_NO_QUOTES)));
         while ($dao->fetch()) {
-          $activities[$dao->activity_id]['case_id'] = $dao->case_id;
+          $activities[$dao->activity_id]['case_id'][] = $dao->case_id;
```

Comment:
If you break something that did not have a unit test, it ain't broken ;)
Question on create, does it work with both case_id =int or case_id=array? Breaking on read is IMO not a huge issue, breaking on create risks loosing data.